sherpa/astro/io/crates_backend.py

```diff
@@ -934,13 +934,13 @@ def get_pha_data(arg, make_copy=True, use_background=False):
 
     if pha is None:
         pha = phadataset.get_crate(phadataset.get_current_crate())
-        if (pha.get_key('HDUCLAS1').value == 'SPECTRUM' or
-                pha.get_key('HDUCLAS2').value == 'SPECTRUM'):
+        if (pha.get_key('HDUCLAS1').value == b'SPECTRUM' or
```

Comment:
I am trying to get the code to work, but yes, in some cases I agree that crates should be returning strings and bury this ugliness down a couple of layers :)

And yes, the refactoring you suggest makes sense either way.